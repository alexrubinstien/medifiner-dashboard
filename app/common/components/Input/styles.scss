@import './app/common/styles/utils/_vars.scss';

.input {
  width: 195px;

  &--wide {
    width: 100%;
  }

  &__label {
    position: relative;
    top: -4px;
    display: inline-block;
    color: $text-color;
    font-size: 12px;
    font-weight: 700;
    line-height: 20px;
    letter-spacing: 1.5px;
    text-transform: uppercase;
  }

  &__input {
    width: 100%;
    padding: 7px 0;
    background: transparent;
    border: 0;
    border-bottom: 2px solid $border-color;
    color: $primary-color;
    font-size: 16px;
    font-weight: 500;
    line-height: 24px;
    letter-spacing: 1px;

    &:-moz-placeholder {
      color: $text-light-color;
    }

    &:-ms-input-placeholder {
      color: $text-light-color;
    }

    &::-webkit-input-placeholder {
      color: $text-light-color;
    }
  }
}
