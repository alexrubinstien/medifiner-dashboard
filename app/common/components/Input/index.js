import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './styles.scss';

const Input = ({
  children,
  className,
  inputClassName,
  label,
  wide,
  ...props,
}) => (
  <div
    className={
      classNames(className, {
        input: true,
        'input--wide': wide,
      })
    }
  >
    <label className="input__label">{label}</label>
    <input
      type="text"
      className={classNames('input__input', inputClassName)}
      {...props}
    />
    {children}
  </div>
);

Input.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  inputClassName: PropTypes.string,
  label: PropTypes.string,
  wide: PropTypes.bool,
};

Input.defaultProps = {
  children: null,
  className: '',
  inputClassName: '',
  label: null,
  wide: false,
};

export default Input;
