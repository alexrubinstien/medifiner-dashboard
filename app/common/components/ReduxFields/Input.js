import React from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';

import Input from '@common/components/Input';

import './styles.scss';

const FormInput = ({
  input,
  label,
  meta,
  ...props
}) => (
  <div className="field">
    <Input
      wide
      id={input.name}
      autoComplete="off"
      label={label}
      {...props}
      {...input}
      inputClassName={classNames('field__input', { 'field__input--error': (meta.touched && meta.error) })}
    />
    {(meta.touched && meta.error) && (
      <div className="field__error">
        {meta.error}
      </div>
    )}
  </div>
);

FormInput.propTypes = {
  input: PropTypes.shape({}).isRequired,
  label: PropTypes.string,
  meta: PropTypes.shape({
    error: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.array,
    ]),
    touched: PropTypes.bool,
  }).isRequired,
};

FormInput.defaultProps = {
  label: null,
};

export default FormInput;
