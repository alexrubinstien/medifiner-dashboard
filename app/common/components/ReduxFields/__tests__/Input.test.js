/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import Input from '../Input.js';

describe('With Snapshot Testing', () => {
  it('Redux input component', () => {
    const mockStore = configureStore([]);
    const store = mockStore();

    const component = renderer.create(
      <Provider store={store}>
        <Input
          label="Label"
          placeholder="Enter Zip Code"
          input={{ name: 'test' }}
          meta={{ touched: false }}
        />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Redux input component with value', () => {
    const mockStore = configureStore([]);
    const store = mockStore();

    const component = renderer.create(
      <Provider store={store}>
        <Input
          label="Label"
          placeholder="Enter Zip Code"
          input={{
            name: 'test',
            value: 'test value',
          }}
          meta={{ touched: true }}
        />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Redux input component with value and error', () => {
    const mockStore = configureStore([]);
    const store = mockStore();

    const component = renderer.create(
      <Provider store={store}>
        <Input
          label="Label"
          placeholder="Enter Zip Code"
          input={{
            name: 'test',
            value: 'test value',
          }}
          meta={{
            touched: true,
            error: 'Test error',
          }}
        />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
