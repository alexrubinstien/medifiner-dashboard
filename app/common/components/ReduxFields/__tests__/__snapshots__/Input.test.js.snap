// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Redux input component 1`] = `
<div
  className="field"
>
  <div
    className="input input--wide"
  >
    <label
      className="input__label"
    >
      Label
    </label>
    <input
      autoComplete="off"
      className="input__input field__input"
      id="test"
      name="test"
      placeholder="Enter Zip Code"
      type="text"
    />
  </div>
</div>
`;

exports[`With Snapshot Testing Redux input component with value 1`] = `
<div
  className="field"
>
  <div
    className="input input--wide"
  >
    <label
      className="input__label"
    >
      Label
    </label>
    <input
      autoComplete="off"
      className="input__input field__input"
      id="test"
      name="test"
      placeholder="Enter Zip Code"
      type="text"
      value="test value"
    />
  </div>
</div>
`;

exports[`With Snapshot Testing Redux input component with value and error 1`] = `
<div
  className="field"
>
  <div
    className="input input--wide"
  >
    <label
      className="input__label"
    >
      Label
    </label>
    <input
      autoComplete="off"
      className="input__input field__input field__input--error"
      id="test"
      name="test"
      placeholder="Enter Zip Code"
      type="text"
      value="test value"
    />
  </div>
  <div
    className="field__error"
  >
    Test error
  </div>
</div>
`;
